### ISSUE TITLE
Check GPU support for shader barycentric coordinates

### ISSUE THREAD

**Issue by billhollings** [KhronosGroup/MoltenVK#1644]
Use `[MVKDevice supportsShaderBarycentricCoordinates]` to set new `MVKPhysicalDeviceMetalFeatures::shaderBarycentricCoordinates`, and use it in `MVKPhysicalDevice::initExtensions()` to disable   `vk_KHR_fragment_shader_barycentric` if set to `false`.

**Comment by spnda** [comment]
This doesn't apply only to this extension. There's multiple extensions that require some Metal GPU Family, some feature tier, or something else. In fact, there's 21 getters on MTLDevice (excluding supportsFamily and supportsFeatureSet) for various features. I feel like MVKExtensions should receive a good refactor as a whole (again!). See the list of supported extensions here:
https://github.com/KhronosGroup/MoltenVK/blob/master/Docs/MoltenVK_Runtime_UserGuide.md#interacting-with-the-moltenvk-runtime

For example, there exists this function:
https://github.com/KhronosGroup/MoltenVK/blob/beee280ff8782b26b96fec9290d82dfd84eded52/MoltenVK/MoltenVK/Layers/MVKExtensions.mm#L47-L81
This function already does a bit of work, yet falsely allows many extensions that would actually not be allowed on certain devices.

My suggestion would be to split up the aforementioned function and then use the functions with the MVK_EXTENSION macro, something like this perhaps:
```cpp
// Will call MVKDevice::supportsBufferAddress and check if the macOS and iOS versions are given to evaluate if it's actually supported.
MVK_EXTENSION(KHR_buffer_device_address,           KHR_BUFFER_DEVICE_ADDRESS,            DEVICE,   12.05,  16.0, supportsBufferAddress)
// Needs no check, universally supported
MVK_EXTENSION(KHR_create_renderpass2,              KHR_CREATE_RENDERPASS_2,              DEVICE,   10.11,  8.0, nullptr)
```

**Comment by billhollings** [comment]
`mvkIsSupportedOnPlatform()` is designed to do a first pass at filtering both instance and device extensions based on the platform and platform version. The `MTLDevice` is not available here and does not apply.

Device extensions are further filtered in `MVKPhysicalDevice::initExtensions()` based on feedback from the settings in `MVKPhysicalDeviceMetalFeatures`.

Your suggestion to use the `MVK_EXTENSION` macro with function calls is an interesting idea, if there are many extensions which need to be filtered. Not sure it it's worth it yet. And if we do go that route, the macro should use values from `MVKPhysicalDeviceMetalFeatures`, not `MTLDevice` functions directly, because the requirement for any extension may be more complicated than a single `BOOL` property access on a `MTLDevice` instance, and the values in `MVKPhysicalDeviceMetalFeatures` are often accessed elsewhere in code.

**Comment by billhollings** [comment]
Fixed in #1649.
